spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/reader/ValueRowReader.scala

```diff
@@ -14,7 +14,10 @@ class ValueRowReader[T: TypeConverter](columnRef: ColumnRef) extends RowReader[T
     * @param row row fetched from Cassandra
     * @param rowMetaData: column names available in the `row` */
   override def read(row: Row, rowMetaData: CassandraRowMetadata): T =
-    converter.convert(GettableData.get(row, columnRef.cqlValueName))
+    converter.convert(GettableData.get(
+      row,
+      columnRef.cqlValueName,
+      rowMetaData.codecs(columnRef.cqlValueName)))
```

Comment:
We can do it even faster.
Now you have to hash table look up per row.
Looks here 
https://github.com/datastax/spark-cassandra-connector/blob/86f0de138f0ce3a11aedd2c400898398a38bcce0/spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/reader/ValueRowReader.scala#L31-L31
Value reader is always read the first column.
and documentation support this statement:
https://github.com/datastax/spark-cassandra-connector/blob/44de124012698ab6bb985e5c45af5eaba4146cac/spark-cassandra-connector/src/main/java/com/datastax/spark/connector/japi/CassandraJavaUtil.java#L346-L346

So you can simple change it to:
converter.convert(GettableData.get(row, 0), rowMetaData.codecs(0))